test_runner/fixtures/utils.py

```diff
@@ -33,9 +34,36 @@
     )
 
     from fixtures.neon_fixtures import PgBin
-from fixtures.common_types import TimelineId
+
 
 Fn = TypeVar("Fn", bound=Callable[..., Any])
+COMPONENT_BINARIES = {
+    "storage_controller": ["storage_controller"],
+    "storage_broker": ["storage_broker"],
+    "compute": ["compute_ctl"],
+    "safekeeper": ["safekeeper"],
+    "pageserver": ["pageserver", "pagectl"],
+}
+
+VERSION_COMBINATIONS = {
+    "argnames": "combination",
+    "argvalues": [
+    {"storage_controller": 'new', "storage_broker":'new', "compute":'new', "safekeeper":'new', "pageserver":'new'},
+    { "storage_controller":'old', "storage_broker":'old', "compute":'old', "safekeeper":'old', "pageserver":'new'},
+    { "storage_controller":'old', "storage_broker":'new', "compute":'old', "safekeeper":'new', "pageserver":'old'},
+    { "storage_controller":'new', "storage_broker":'old', "compute":'new', "safekeeper":'old', "pageserver":'old'},
+    { "storage_controller":'new', "storage_broker":'old', "compute":'old', "safekeeper":'new', "pageserver":'old'},
+    { "storage_controller":'old', "storage_broker":'new', "compute":'new', "safekeeper":'old', "pageserver":'old'}
+],
+    "ids": [
+        "combination_nnnnn",
+        "combination_oooon",
+        "combination_onono",
+        "combination_nonoo",
+        "combination_noono",
+        "combination_onnoo",
+    ]
+}
```

Comment:
Let's keep this as a function and generate `ids` automatically to make it easier to add more combinations.
Also, let's add a comment with an explanation of what's going on here (the function generates parameters for `@pytest.mark.parametrize`, why we choose only this set of parameters, etc)

Reply:
done